src/ccl_background.c

```diff
@@ -1013,6 +1013,68 @@ void ccl_comoving_angular_distances(ccl_cosmology * cosmo, int na, double a[],
   }
 }
 
+
+double ccl_comoving_angular_diameter_distance(ccl_cosmology * cosmo, double a1, double a2, int* status)
+{
+  if(a1>1. || a2>1.) {
+    *status = CCL_ERROR_COMPUTECHI;
+    ccl_raise_warning(*status,"CCL_ERROR_COMPUTECHI: scale factor cannot be larger than 1.");
+    return NAN;
+  } else {
+    if(cosmo->params.Omega_k<0.){
+      *status = CCL_ERROR_COMPUTECHI;
+      ccl_raise_warning(*status,"CCL_ERROR_COMPUTECHI: Omega_k cannot be negative for angular diameter distance.");
+      return NAN;
+    } else {
+      if (!cosmo->computed_distances) {
+	*status = CCL_ERROR_DISTANCES_INIT;
+	ccl_cosmology_set_status_message(cosmo,"ccl_background.c: ccl_h_over_h0(): distance splines have not been precomputed!");
```

Comment:
This error message is for the wrong function.

Reply:
Done